### DISCUSSION TITLE
Bluetooth isochronous channels central_iso / peripheral_iso example not working reliable on nrf52840dk – packets not received/lost

### DISCUSSION THREAD

**Discussion by Benni77** [zephyrproject-rtos/zephyr#70754] [Q&A]
**EDIT2:**
ISSUE: [#70794](https://github.com/zephyrproject-rtos/zephyr/issues/70794)

**EDIT:** Using the nrf connect sdk 2.6.0, it appears that the examples run without any problems. What does this mean for the zephyr repo (esp. Zephyr BLE Controller)?

...

**Zephyr Version: 3.6.99**

Hey guys,

my goal is to transfer data from nrf52832 to nrf52840 via BLE ISO channel. Since my own implementation did not work as expected (Packets not received) I thought I give the examples a try: **[CENTRAL_ISO](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/bluetooth/central_iso)** & **[PERIPHERAL_ISO](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/bluetooth/peripheral_iso)**

Unfortunately, I had to realize that the same problem occurs with the examples, both examples run on **nrf52840** for now. What happens:
1. Connection is established between central and peripheral.
2. ISO channel is connected succesfully.
3. Central starts data stream (all parameters from example left unchanged)
4. Peripheral receives some data and prints messages to console.
5. At some point (after few intervals) the peripheral is not receiving any more data, while central is still sending?

I checked with Segger's Sysview and saw that the BT TX thread is called periodically on the central, but the BT RX thread is not called on the peripheral after data reception has stopped.

I can improve reliability by increasing the interval, which results in a longer data stream before it is stopped.

Does anyone know a fix to this problem? Seems like a bug to me, because nrf52840 should support iso channels, right?

Thank you!

**Comment by koffes** [comment]
Hi @Benni77.
Let me shed some light on the connection between NCS and Zephyr.

First, these are Zephyr samples you are referring to, so asking here (as you did) or in the Zephyr Discord channel are good places to start.

NCS X.X.X uses a given SHA of Zephyr with some changes. The Zephyr version used by NCS is located here: https://github.com/nrfconnect/sdk-zephyr and the exact version can be found in the NCS manifest https://github.com/nrfconnect/sdk-nrf/blob/main/west.yml#L66-L76.

Furthermore, NCS Bluetooth samples defaults to Nordic's Softdevice Controller, whilst Zephyr uses the open source Zephyr controller. 

With that in mind, mentioning NCS here may raise questions for some, as NCS depends on Zephyr, but not the other way around.
If you do need assistance with Nordic devices, please reach out on https://devzone.nordicsemi.com/.
